### PR TITLE
[MISC] Add README.md file

### PR DESCRIPTION
**Pull request by deusebio**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,71 @@
...
+for developer guidance. 
+
+### We are Hiring!
+
+Also, if you truly enjoy working on open-source projects like this one and you 
+would like to be part of the OSS revolution, please don't forget to check out 
+the [open positions](https://canonical.com/careers/all) we have at [Canonical](https://canonical.com/). 
+
+## License
+The Charmed Kafka Operator is free software, distributed under the
```
Comment (welpaolo) [inline README.md:68]:
```suggestion
The Spark Integration Hub charm is free software, distributed under the 
```

File: README.md [hunk 2]
```diff
@@ -1 +1,71 @@
...
+juju add-model <my-model>
+juju deploy s3-integrator --channel latest/edge
+juju deploy spark-integration-hub-k8s --channel edge
+juju relate spark-integration-hub-k8s s3-integrator
+```
+
+When creating new Spark service account using the [`spark-client` snap](https://snapcraft.io/spark-client)
+
+```shell
+spark-client.service-account-registry create --username <spark-user> --namespace <my-model>
```
Comment (welpaolo) [inline README.md:44]:
```suggestion
spark-client.service-account-registry create --username <spark-user> --namespace <namespace>
```

File: README.md [hunk 3]
```diff
@@ -1 +1,71 @@
...
+```
+
+The Spark Integration Hub will take care of adding relevant configuration to the 
+Charmed Spark properties, 
+
+```shell
+spark-client.service-account-registry get-config --username <spark-user> --namespace <my-model>
```
Comment (welpaolo) [inline README.md:51]:
```suggestion
spark-client.service-account-registry get-config --username <spark-user> --namespace <namespace>
```

File: README.md [hunk 4]
```diff
@@ -1 +1,71 @@
...
+
+* Handle relations with other charms and translate them in configuration options 
+  that Spark can handle (like the Spark History Server charm does with the S3-integrator).
+* Store the configuration options in Kubernetes secrets for the service account
+  managed by the Charmed Spark solution (all service account are labeled with predefined labels)
+* Handle the addition/removal of service accounts with corresponding updates of 
+  configurations.
+* Handle updates from the related charms (i.e., new database credentials, 
+  change of endpoints, updated S3 credentials.)
+* The configuration hub charm may also be able to create and configure services
```
Comment (welpaolo) [inline README.md:18]: This feature is not available yet

**Review by welpaolo — APPROVED** [review]
LGMT, I left some comments.

File: README.md (hunk 4 above)
Comment (deusebio) [reply README.md:18]: But we provide the interface already [here](https://github.com/canonical/spark-integration-hub-k8s-operator/blob/main/metadata.yaml#L42), don't we?

File: README.md (hunk 2 above)
Comment (deusebio) [reply README.md:44]: using `my-model` at line `35`

Comment (welpaolo) [reply README.md:44]: yeah, I see. What I was trying to say is that a user is not forced to use the same namespace as the juju model.

Comment (deusebio) [reply README.md:44]: That's fine then!

**Review by theoctober19th — APPROVED** [review]

**Pull request merged**